app/Platform/Actions/UnlockPlayerAchievement.php

```diff
@@ -77,7 +77,9 @@ public function execute(
         }
 
         // commit the unlock
-        $unlock->save();
+        if ($achievement->is_published) {
+            $unlock->save();
+        }
 
         // post the unlock notification
         PlayerAchievementUnlocked::dispatch($user, $achievement, $hardcore);
```

Comment:
Recommend stifling the event as well. If the unlock wasn't committed, it didn't really happen. There's no reason to rebuild the player, game, or developer contribution metrics, and anything else that might be watching won't be able to find the unlock if they query for it.

Reply:
No problem, done in latest.